contracts/SDUtilityPool.sol

```diff
@@ -0,0 +1,771 @@
+        accrueFee();
+        return _exchangeRateStoredInternal();
+    }
+
+    /**
+     * @dev Triggers stopped state.
+     * Contract must not be paused
+     */
+    function pause() external {
+        UtilLib.onlyManagerRole(msg.sender, staderConfig);
+        _pause();
+    }
+
+    /**
+     * @dev Returns to normal state.
+     * Contract must be paused
+     */
+    function unpause() external onlyRole(DEFAULT_ADMIN_ROLE) {
+        _unpause();
+    }
+
+    //Setters
+
+    /**
+     * @notice updates protocol fee factor
+     * @dev only `MANAGER` role can call
+     * @param _protocolFeeFactor value of protocol fee factor
+     */
+    function updateProtocolFeeFactor(uint256 _protocolFeeFactor) external override {
+        UtilLib.onlyManagerRole(msg.sender, staderConfig);
+        protocolFeeFactor = _protocolFeeFactor;
+        emit ProtocolFeeFactorUpdated(protocolFeeFactor);
+    }
+
+    /**
+     * @notice updates the utilization rate
+     * @dev only `MANAGER` role can call
+     * @param _utilizationRatePerBlock new value of utilization rate per block
+     */
+    function updateUtilizationRatePerBlock(uint256 _utilizationRatePerBlock) external override {
+        UtilLib.onlyManagerRole(msg.sender, staderConfig);
+        utilizationRatePerBlock = _utilizationRatePerBlock;
+        emit UtilizationRatePerBlockUpdated(utilizationRatePerBlock);
+    }
+
+    /**
+     * @notice updates the maximum amount ETH worth of SD utilize per validator
+     * @dev only `MANAGER` role can call
+     * @param _maxETHWorthOfSDPerValidator new value of maximum amount ETH worth of SD utilize per validator
+     */
+    function updateMaxETHWorthOfSDPerValidator(uint256 _maxETHWorthOfSDPerValidator) external override {
+        UtilLib.onlyManagerRole(msg.sender, staderConfig);
+        maxETHWorthOfSDPerValidator = _maxETHWorthOfSDPerValidator;
+        emit UpdatedMaxETHWorthOfSDPerValidator(_maxETHWorthOfSDPerValidator);
+    }
+
+    /**
+     * @notice updates the batch limit to finalize withdraw request in a single txn
+     * @dev only `MANAGER` role can call
+     * @param _finalizationBatchLimit new value of batch limit
+     */
+    function UpdateFinalizationBatchLimit(uint256 _finalizationBatchLimit) external override {
+        UtilLib.onlyManagerRole(msg.sender, staderConfig);
+        finalizationBatchLimit = _finalizationBatchLimit;
+        emit UpdatedFinalizationBatchLimit(finalizationBatchLimit);
+    }
+
+    /**
+     * @notice updates the undelegation period of a withdraw request
+     * @dev only `DEFAULT_ADMIN_ROLE` role can call
+     * @param _undelegationPeriodInBlocks new value of undelegationPeriodInBlocks
+     */
+    function updateUndelegationPeriodInBlocks(uint256 _undelegationPeriodInBlocks)
+        external
+        override
+        onlyRole(DEFAULT_ADMIN_ROLE)
+    {
+        undelegationPeriodInBlocks = _undelegationPeriodInBlocks;
```

Comment:
Better to ensure that this is > than maxFinalizationDelay.

Same for the finalizationDelay setter.